specs-go/chain/chainid.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2016 The Linux Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package chain provides implementations of the ChainID calculation used in
+// identifying the result of subsequent layer applications.
+package chain
+
+import "github.com/docker/distribution/digest"
```

Comment:
Yes, I'd be strongly in favour of moving it in here so that any spec implementers can/will easily leverage it.